### PR TITLE
Uncertainty regression calibrators (non-conformal) 

### PR DESCRIPTION
**Pull request by shihchengli**
## Description
Work in progress to implement:
- zscaling
- zelikman-interval
- mve_weighting

tscaling will not be implemented.

## Questions
The `mve_weighting` hasn't been added to this PR yet. The main reason is that this method requires the predicted variance for each individual model in an ensemble. This method minimizes the negative log-likelihood for the predictions versus the targets by applying a weighted average across the ensemble's variance predictions (https://doi.org/10.1186/s13321-021-00551-x).

To support this, we would need to modify the docstring and signature for this class, as well as for the uncertainty predictors of MVE and evidential methods, since only those two methods can predict variance for each model in an ensemble. This might make the code a bit messy because the signature defined in the base class may not necessarily be used to the child class.
Specifically, we would need the shape of uncertainty returned from `MVEPredictor` and `EvidentialPredictor` be ``m x n x t`` instead of ``n x t``, where ``m`` is the number of models, ``n`` is the number of inputs, ``t`` is the number of tasks. The input of `uncs` in the `fit()` of `MVEWeightingCalibrator` method would also be in the  same shape of ``m x n x t``.

EDIT: I have followed @KnathanM's suggestion (https://github.com/chemprop/chemprop/pull/1055#issuecomment-2419800352) to implement `mve_weighting`.

## Checklist
- [x] linted with flake8?
- [x] (if appropriate) unit tests added?

### CODE REVIEW

**Comment by shihchengli** [comment]
The `zscaling` and `zelikman-interval` is ready for review. Before I work on the `mve_weighting` method, I would appreciate any input for the question mentioned above.

**Comment by KnathanM** [comment]
We could change the dimension of uncertainty model outputs to be:
`preds -> m x n x t`, where `m` is the number of predictions (the number of models in the original list).
`uncs -> m' x n x t` where `m'` is the number of uncertainty estimations. For two MVE models, this would be two, but for an ensemble of two models, this would be one (even though it produces `m = 2` because it has two target predictions).

For an ensemble of MVE models, you would get `m=m'` and we don't average the uncertainty predictions. If the user wants to use the average of the uncertainties as the final uncertainty value (as we will probably do in the CLI), they can average them afterwards.

File: chemprop/uncertainty/calibrator.py [hunk 1]
```diff
@@ -76,38 +76,148 @@ def fit(self, preds: Tensor, uncs: Tensor, targets: Tensor, mask: Tensor) -> Sel
...
         return self
 
     def apply(self, uncs: Tensor) -> Tensor:
-        return
+        return uncs * self.scalings**2
 
 
 @UncertaintyCalibratorRegistry.register("zelikman-interval")
 class ZelikmanCalibrator(RegressionCalibrator):
+    """Calibrate regression datasets that does not depend on a particular probability function form.
```
Comment (KnathanM) [inline chemprop/uncertainty/calibrator.py:114]:
```suggestion
    """Calibrate regression datasets using a method that does not depend on a particular probability function form.
```

File: chemprop/uncertainty/calibrator.py [hunk 2]
```diff
@@ -76,38 +76,148 @@ def fit(self, preds: Tensor, uncs: Tensor, targets: Tensor, mask: Tensor) -> Sel
...
+
+    This method is designed to be used with interval output. Uses the "CRUDE" method as described in [zelikman2020]_.
```
Comment (KnathanM) [inline chemprop/uncertainty/calibrator.py:116]: can you add that this currently only works for variance?

Comment (shihchengli) [reply chemprop/uncertainty/calibrator.py]: added

**Review by KnathanM — APPROVED** [review]

**Pull request merged**